server/src/main/java/org/apache/cassandra/sidecar/tasks/KeyStoreCheckPeriodicTask.java

```diff
@@ -41,25 +41,49 @@ public class KeyStoreCheckPeriodicTask implements PeriodicTask
     private static final Logger LOGGER = LoggerFactory.getLogger(KeyStoreCheckPeriodicTask.class);
 
     private final Vertx vertx;
-    private final Provider<Server> server;
-    private final SslConfiguration configuration;
+    private final SslConfiguration sslConfiguration;
+    private final Function<Long, Future<Boolean>> updateSSLOptionsFunction;
     private long lastModifiedTime = 0; // records the last modified timestamp
+    private final String taskName;
 
-    public KeyStoreCheckPeriodicTask(Vertx vertx, Provider<Server> server, SidecarConfiguration configuration)
+    public KeyStoreCheckPeriodicTask(Vertx vertx,
+                                     SslConfiguration sslConfiguration,
+                                     Function<Long, Future<Boolean>> updateSSLOptionsFunction,
+                                     String taskName)
     {
         this.vertx = vertx;
-        this.server = server;
-        this.configuration = configuration.sslConfiguration();
+        this.sslConfiguration = sslConfiguration;
+        this.updateSSLOptionsFunction = updateSSLOptionsFunction;
+        this.taskName = taskName;
+    }
+
+    public static KeyStoreCheckPeriodicTask forServer(Vertx vertx,
+                                                      SslConfiguration sslConfiguration,
+                                                      Function<Long, Future<Boolean>> updateSSLOptionsFunction)
+    {
+        return new KeyStoreCheckPeriodicTask(vertx, sslConfiguration, updateSSLOptionsFunction, "ServerKeyStoreCheckPeriodicTask");
+    }
+
+    public static KeyStoreCheckPeriodicTask forClient(Vertx vertx,
+                                                      SslConfiguration sslConfiguration,
+                                                      Function<Long, Future<Boolean>> updateSSLOptionsFunction)
+    {
+        return new KeyStoreCheckPeriodicTask(vertx, sslConfiguration, updateSSLOptionsFunction, "ClientKeyStoreCheckPeriodicTask");
+    }
+
+    @Override
+    public String name()
+    {
+        return taskName;
     }
 
     @Override
     public void deploy(Vertx vertx, PeriodicTaskExecutor executor)
     {
-        if (configuration != null
-            && configuration.enabled()
-            && configuration.keystore() != null
-            && configuration.keystore().isConfigured()
-            && configuration.keystore().reloadStore())
+        if (sslConfiguration != null &&
+            sslConfiguration.keystore() != null
+            && sslConfiguration.keystore().isConfigured()
+            && sslConfiguration.keystore().reloadStore())
```

Comment:
I think we should still check for the configuration being enabled here, any reason to remove the check?
```suggestion
        if (sslConfiguration != null
            && configuration.enabled()
            && sslConfiguration.keystore() != null
            && sslConfiguration.keystore().isConfigured()
            && sslConfiguration.keystore().reloadStore())
```